File: tfc_gym_app/lib/data/models/dto/routine_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'routine_dto.g.dart';

@JsonSerializable()
class RoutineDTO {
  final int routineId;
  final String? routineName;
  final String? routineDescription;
  final int userId;

  RoutineDTO({
    required this.routineId,
    this.routineName,
    this.routineDescription,
    required this.userId,
  });

  factory RoutineDTO.fromJson(Map<String, dynamic> json) => _$RoutineDTOFromJson(json);
  Map<String, dynamic> toJson() => _$RoutineDTOToJson(this);
}

File: service/TFC.Application.DTO/Entity/RoutineDTO.cs
namespace TFC.Application.DTO.Entity
{
    public class RoutineDTO
    {
        public long RoutineId { get; set; }
        public string? RoutineName { get; set; }
        public string? RoutineDescription { get; set; }
        public long UserId { get; set; }
        public List<SplitDayDTO> SplitDays { get; set; } = new List<SplitDayDTO>();
    }
}